cardano-node/src/Cardano/Node/Handlers/Shutdown.hs

```diff
@@ -81,6 +83,13 @@ parseShutdownOn = asum
     ]
   , pure NoShutdown
   ]
+  where
+    bounded :: forall a. (Bounded a, Integral a, Show a) => String -> Opt.ReadM a
+    bounded t = Opt.eitherReader $ \s -> do
+      i <- Read.readEither @Integer s
+      when (i < fromIntegral (minBound @a)) $ Left $ t <> " must not be less than " <> show (minBound @a)
+      when (i > fromIntegral (maxBound @a)) $ Left $ t <> " must not greater than " <> show (maxBound @a)
+      pure (fromIntegral i)
```

Comment:
Is this needed?

Reply:
It's gone from cardano-api

Reply:
It exists in `Cardano.CLI.EraBased.Options.Common` but it's not exposed.